### DISCUSSION TITLE
Is it possible to "deregister" a loader hook?

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [nodejs/help#4614] [Q&A] [answered]
```ts
import { register } from 'node:module'

export function registerLoader() {
  register('./loader.mjs', import.meta.url)
}

export function deregisterLoader() {
  // ...
}
```

Is it possible to deregister a specific hook?

**Comment by aduh95** [comment, marked as answer]
There was a PR opened for this, but it looks like it has kinda stalled: https://github.com/nodejs/node/pull/49159

Currently it's not possible. What would be your use-case for this?

**Reply by devjiwonchoi** [reply]
I need to resolve a specific TypeScript file (`config.ts`) and it's imports.
This config file will be loaded initially, so I'm worried about perf as the loader remains the rest of the process, which is unnecessary.

**Comment by hernandemonteiro** [comment]
In Node.js, as of now, there's no built-in API to deregister a specific module loader hook that was registered with register() from the 'module' package. Once a loader is registered, it remains active for the entire lifecycle of the process, and there's no official method to undo this or remove a specific hook.

If you need a workaround, here are a few approaches to consider:

1. Use Conditional Registration
Rather than deregistering, you could conditionally register the loader based on certain criteria. For instance, you might control the conditions under which registerLoader is called so that it's only set once or under specific conditions.

2. Isolate Loader Usage
If you want different behaviors without "unregistering" loaders, consider running the code in isolated processes or worker threads, where each instance can be started with or without specific loaders as needed.

3. Restart the Process (Last Resort)
If the loader affects a short-lived process or test environment, restarting the process is sometimes used to clear module states, including loader hooks.

Future Support
Node.js may support deregistering loaders in the future, so it’s worth checking periodically for updates in the Node.js documentation.
